CRM/Contact/Import/Parser/Contact.php

```diff
@@ -122,11 +122,11 @@ class CRM_Contact_Import_Parser_Contact extends CRM_Contact_Import_Parser {
    * @param array $mapperRelatedContactWebsiteType
    */
   public function __construct(
-    &$mapperKeys, $mapperLocType = [], $mapperPhoneType = [], $mapperImProvider = [], $mapperRelated = [], $mapperRelatedContactType = [], $mapperRelatedContactDetails = [], $mapperRelatedContactLocType = [], $mapperRelatedContactPhoneType = [], $mapperRelatedContactImProvider = [],
```

Comment:
@eileenmcnaughton are you sure we don't need to modify the MapperKeys?

Reply:
I'm pretty sure - I've been digging around these classes a bit & it seems to just be a phpv4 hangover - we've removed the & from a bunch already

Reply:
To work backwards - the calling class would need to *do something* to change them for this to make sense - but actually it is just a list of field names submitted on the form

Reply:
@seamuslee001  Specifically it's called  from

**CRM_Contact_Import_Form_MapField::submit**  
uses a variable   to pass in & not  again

**CRM_Contact_Import_Form_DataSource::postProcess**
Creates empty array  to pass to constructor & then (as  a  non-reference) to run - just 2  lines later  & not again

**CRM_Contact_Import_Form_Preview::postProcessOld**
This is never called.....

 CRM_Contact_Import_ImportJob::runImport
The value is not used again.

Also this one of the calls to this  function is called from runIncompleteImportJobs - which is never called....